### ISSUE TITLE
Investigate pros and cons of Postgres upgrade

### ISSUE THREAD

**Issue by sauloperez** [openfoodfoundation/ofn-install#272]
At this point not using an up-to-date version such as Postgres 10 (11 is probably still too new), we miss several performance and stability improvements we could benefit from. 

It is a higher priority however to upgrade the latest patch of our current version. We're currently using 9.5.14 and yesterday 9.5.15 got released. You can check it out at https://www.postgresql.org/about/news/1905/.

Upgrading the database might be a tricky process although Postgres provides [pg_upgrade](https://www.postgresql.org/docs/9.5/pgupgrade.html) to do that. We need to find out the pros and cons an any possible consequences of doing it.

**Comment by sauloperez** [comment]
https://why-upgrade.depesz.com/show?from=9.5&to=11.1 comes in handy here.

**Comment by Matt-Yorkley** [comment]
I think postgres 10 will only be available when we transition to Bionic Beaver? 

Also, I think our current Rails version is keeping us stuck to < 10 through gem dependencies.

**Comment by sauloperez** [comment]
> Also, I think our current Rails version is keeping us stuck to < 10 through gem dependencies.

that's a valid point but even at < 10 we would get a lot.

**Comment by Matt-Yorkley** [comment]
I've just realised we don't specify the postgres version anywhere in ofn-install, so it's just installing whatever version the OS decides. That's probably not the best. I'd have to check, but i think this means a fresh Bionic Beaver install might use postgres 10...?

**Comment by sauloperez** [comment]
> I've just realised we don't specify the postgres version anywhere in ofn-install

There might even be an old issue here about it

> but i think this means a fresh Bionic Beaver install might use postgres 10...?

Good catch 🙈 . AFAIK we're using 9.5 at least that is what I set in the dev setup script I did for OFN.

**Comment by Matt-Yorkley** [comment]
Well I've already booted OFN in a fresh Bionic Beaver box a couple of times and tested it a bit. I didn't get any immediate errors. This obviously needs more investigating...

**Comment by sauloperez** [comment]
Time to provision Katuma staging with it?

**Comment by Matt-Yorkley** [comment]
Should we take a look at #99, #72, and #227 first? Or just go for it and see what happens? It's up to you.

Anem! :muscle:

**Comment by sauloperez** [comment]
What about both? We should clean up regardless of Bionic.

**Comment by luisramos0** [comment]
I think we can close this spike now and just upgrade all servers to bionic with pg 10 :tada: 

AUS
Description:	Ubuntu 18.04.4 LTS
psql (PostgreSQL) 10.12 (Ubuntu 10.12-0ubuntu0.18.04.1)

UK
Description:	Ubuntu 16.04.5 LTS
psql (PostgreSQL) 9.5.19

FR:
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
psql (PostgreSQL) 9.5.19

**Comment by sauloperez** [comment]
YES! Upgrading the recent PostgreSQL really sounds like a way to get performance for free.

**Comment by Matt-Yorkley** [comment]
I just took a quick survey with `ansible all-prod -u ofn-admin -a "apt-cache policy postgresql"` and we have:

- US, AUS, and Katuma on Postgres 10
- Everyone else on Postgres 9.5

I'm pretty sure I could knock up a playbook that handles this in no time (using the `pg_upgrade` option) and we could automate the whole process and just upgrade them all at once. It would be easy to test in staging, and if it works we could even try upgrading all servers from 10 to 11 with the same strategy.

**Comment by Matt-Yorkley** [comment]
Update; the playbook looks like this: https://github.com/openfoodfoundation/ofn-install/pull/733

**Comment by mkllnk** [comment]
This is outdated. All mentioned servers have been replaced with newer versions.
